chardet/utf8prober.py

```diff
@@ -75,7 +75,11 @@ def feed(self, byte_str):
 
     def get_confidence(self):
         unlike = 0.99
-        if self._num_mb_chars < 6:
+        if self._num_mb_chars == 1:
+            # guaranteed to be preferred over ISO-8859-1, evaluates to 0.731
+            unlike *= self.ONE_CHAR_PROB * (1-0.731)/0.99/0.5
```

Comment:
While I completely agree that making the default UTF-8 instead of ISO-8859-1 makes a lot of sense (since the world has changed dramatically since the original C code chardet is a port of was written 25+ years ago), I'm not sure that this is the best approach. First of all, it has a rather confusing equation in it that yields a constant value, instead of just returning the constant value. Second, it's not clear to most people while 0.731 will make it win out of ISO-8859-1. Finally, we should probably just reduce the confidence of the ISO-8859-1 prober (or modify what `ONE_CHAR_PROB` is here).

Reply:
Thank you for the feedback.
We would have to reduce the maximum confidence for ISO-8859-1 to 50%. E.g. in Germany you will run into ISO-8859-1 everyday (6% of websites and legacy software).


My motivation was to still allow the user to set ONE_CHAR_PROB. Also for 1 char and not change the behavior for 2+ chars.
Agreed it should be simpler and the comment is confusing.
Do you have some context which qualifies the confidence jump from 1 to 2 chars (50,5% to 75,25%)? In between most other models have there normal confidence levels and get preferred.
If not i would propose a simpler solution:
```python3
# Prefer UTF-8 over other models, can be modified with ONE_CHAR_PROB.
if self._numb_chars == 1:
    unlike *= self.ONE_CHAR_PROB * 1.99
```
above the magic numbers of 73 and 75 and smaller than 2 chars. And easy to read.
Or another alternative just change the original code to max(self._numb_chars, 2).

The equivalent for other models seems to be this, which has some inheritance ongoing:
https://github.com/chardet/chardet/blob/eb1a10a5d1b63bd9aefd1df9c669c298277e420b/chardet/chardistribution.py#L92-L93